g2p_bank/tests/test_bank.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import random
+
+from schwifty import IBAN
+
+from odoo.tests import tagged
+from odoo.tests.common import TransactionCase
+
+_logger = logging.getLogger(__name__)
+
+
+@tagged("post_install", "-at_install")
+class BankTest(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super(BankTest, cls).setUpClass()
+
+        # Initial Setup of Variables
+        cls.registrant_1 = cls.env["res.partner"].create(
+            {
+                "name": "Heidi Jaddranka",
+                "family_name": "Jaddranka",
+                "given_name": "Heidi",
+                "is_group": False,
+                "is_registrant": True,
+            }
+        )
+
+    def test_01_add_bank(self):
+        # Search for Sample Bank or Create if None
+        bank = self.env["res.bank"].search([("name", "=", "Sample Bank")])
+        bank_id = None
+        if bank:
+            bank_id = bank[0]
+        else:
+            vals = {"name": "Sample Bank", "bic": "1010101010"}
+            bank_id = self.env["res.bank"].create(vals)
+
+        vals = []
+
+        # Generate Number Account Number
+        random_acc = str(random.randint(1, 9999999999))
+
+        # Compute Expected IBAN
+        expected_iban = IBAN.generate(
+            bank_id.country.code,
+            bank_code=bank_id.bic,
+            account_code=random_acc,
+        )
```

Comment:
Do not use the same function as you are testing to generate the expected result. hardcode the expected result.

Reply:
Okay